plugin.video.drnu/resources/lib/addon.py

```diff
@@ -0,0 +1,508 @@
+#  http://www.gnu.org/copyleft/gpl.html
+#
+import pickle
+import os
+import sys
+import urlparse
+import re
+import datetime
+
+import xbmc
+import xbmcgui
+import xbmcaddon
+import xbmcplugin
+
+from resources.lib import tvapi
+from resources.lib import tvgui
+
+import buggalo
+
+tr = xbmcaddon.Addon().getLocalizedString
+get_setting = xbmcaddon.Addon().getSetting
+addon_path = xbmcaddon.Addon().getAddonInfo('path')
+addon_name = xbmcaddon.Addon().getAddonInfo('name')
+
+def make_notice(object):
+    xbmc.log(str(object), xbmc.LOGNOTICE )
+
+class DrDkTvAddon(object):
+    def __init__(self, plugin_url, plugin_handle):
+        self._plugin_url = plugin_url
+        self._plugin_handle = plugin_handle
+        # PATH = sys.argv[0]
+        # HANDLE = int(sys.argv[1])
+
+        self.cache_path = xbmc.translatePath(xbmcaddon.Addon().getAddonInfo('profile'))
+        if not os.path.exists(self.cache_path):
+            os.makedirs(self.cache_path)
+
+        self.favorites_path = os.path.join(self.cache_path, 'favorites.pickle')
+        self.recent_path = os.path.join(self.cache_path, 'recent.pickle')
+        self.fanart_image = os.path.join(addon_path, 'resources', 'fanart.jpg')
+
+        self.api = tvapi.Api(self.cache_path)
+        self.favorites = list()
+        self.recentlyWatched = list()
+
+        self.menuItems = list()
+        runScript = "RunAddon(plugin.video.drnu,?show=areaselector&random=%d)" % self._plugin_handle
+        self.menuItems.append((tr(30511), runScript))
+
+
+    def _save(self):
+        # save favorites
+        self.favorites.sort()
+        pickle.dump(self.favorites, open(self.favorites_path, 'wb'))
+
+        self.recentlyWatched = self.recentlyWatched[0:25]  # Limit to 25 items
+        pickle.dump(self.recentlyWatched, open(self.recent_path, 'wb'))
+
+    def _load(self):
+        # load favorites
+        if os.path.exists(self.favorites_path):
+            try:
+                self.favorites = pickle.load(open(self.favorites_path, 'rb'))
+            except Exception:
+                pass
+
+        # load recently watched
+        if os.path.exists(self.recent_path):
+            try:
+                self.recentlyWatched = pickle.load(open(self.recent_path, 'rb'))
+            except Exception:
+                pass
+
+    def showAreaSelector(self):
+        gui = tvgui.AreaSelectorDialog()
+        gui.doModal()
+        areaSelected = gui.areaSelected
+        del gui
+
+        if areaSelected == 'none':
+            pass
+        elif areaSelected == 'drtv':
+            self.showMainMenu()
+        else:
+            items = self.api.getChildrenFrontItems('dr-' + areaSelected)
+            #xbmc.executebuiltin('Container.SetViewMode(500)')
+            self.listSeries(items)
+
+    def showMainMenu(self):
+        items = list()
+        # Live TV
+        item = xbmcgui.ListItem(tr(30027), iconImage=os.path.join(addon_path, 'resources', 'icons', 'livetv.png'))
```

Comment:
please remove the iconimage and the setproperty(fanart image) both are deprecated.
Use:

`item.setArt({"icon":"xxx", "fanart": "xxx"})`